### DISCUSSION TITLE
tutorials/federation-typescript/06-local-rover-dev

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#142] [course feedback]
# tutorials/federation-typescript/06-local-rover-dev

https://www.apollographql.com/tutorials/federation-typescript/06-local-rover-dev

**Comment by nijhoomroy** [comment]
Hey, I am stuck at this step:

<MyComputer>-MBP router % rover dev --supergraph-config ./supergraph-config.yaml
downloading the 'router' plugin from https://rover.apollo.dev/tar/router/x86_64-apple-darwin/latest-plugin
error: error decoding response body: operation timed out

Anyone knows how to fix this?

**Reply by nijhoomroy** [reply]
It worked out though

**Reply by mabuyo** [reply]
Good to hear it @nijhoomroy - what was the fix?

**Comment by hcone5006** [comment]
When I try to run rover using the following command:

"rover dev --supergraph-config ./router/supergraph-config.yaml --supergraph-port 4002"

I get the following error, but when I check, there is nothing running on port 4002? I also have no luck trying to run this command using another port... I get the same error. I saw somewhere a suggestion of updating rover to the latest version but this also did nothing to fix this error.

"WARN: Connector debugging is enabled, this may expose sensitive information.
==> Your supergraph is running! head to http://localhost:4002 to query your supergraph
ERROR: could not create the HTTP server: Address already in use (os error 48)
ERROR: could not create the HTTP server: Address already in use (os error 48)
Router process exited with status code: 1

Router binary exited, stopping `rover dev` processes..."

**Reply by Hiipivahalko** [reply]
I also got this error and tried with different ports but it didn't work. This happens because I was still running the `router` on my other terminal window. Those both use same port for health checks (posrt 8088). The tutorial could be updated to for shutting down the router when using `rover dev`.

**Reply by lizhennessy** [reply]
Thanks for the feedback you two! I will take a look at the course and make the update!

**Comment by MarkAajuLarsen** [comment]
`router/supergraph-config.yaml`

Should maybe be 

```diff
federation_version: =2.7.0
subgraphs:
  listings:
    routing_url: http://localhost:4000
    schema:
-      file: ./listings/src/schema.graphql
+      file: ../listings/src/schema.graphql
  reviews:
    routing_url: http://localhost:4001
    schema:
-      file: ./reviews/src/schema.graphql
+      file: ../reviews/src/schema.graphql
```
Otherwise running `rover dev --supergraph-config ./router/supergraph-config.yaml --supergraph-port 4002` from the root of the project will result in: (At least it did on my machine, might be related to editor settings) 

```
error: Error occurred when composing supergraph
Error resolving subgraphs:
Unable to resolve subgraphs.
listings: Could not find schema file (./listings/src/schema.graphql) relative to (/odyssey-federation-typescript/router) for subgraph `listings`
reviews: Could not find schema file (./reviews/src/schema.graphql) relative to (/odyssey-federation-typescript/router) for subgraph `reviews`
```
